app/config/errors.php

```diff
@@ -519,6 +546,46 @@
         'description' => 'The project key has expired. Please generate a new key using the Appwrite console.',
         'code' => 401,
     ],
+    Exception::ROUTER_INVALID_TYPE => [
+        'name' => Exception::ROUTER_INVALID_TYPE,
+        'description' => 'Invalid domain configuration. Route type is not supported.',
+        'code' => 400,
+    ],
+    Exception::ROUTER_UNKNOWN_HOST => [
+        'name' => Exception::ROUTER_UNKNOWN_HOST,
+        'description' => 'Host is not trusted. Add a custom domain to your project first.',
+        'code' => 400,
+    ],
+    Exception::RULE_RESOURCE_ID_MISSING => [
+        'name' => Exception::RULE_RESOURCE_ID_MISSING,
+        'description' => 'With resourceType you provided, the resourceId is required.',
+        'code' => 400,
+    ],
+    Exception::RULE_CONFIGURATION_MISSING => [
+        'name' => Exception::RULE_CONFIGURATION_MISSING,
+        'description' => '_APP_DOMAIN_TARGET must be a public domain.',
+        'code' => 501,
+    ],
+    Exception::RULE_RESOURCE_ID_NOT_FOUND => [
+        'name' => Exception::RULE_RESOURCE_ID_NOT_FOUND,
+        'description' => 'Resource could not be found. Check resourceId and resourceType.',
+        'code' => 404,
+    ],
+    Exception::RULE_NOT_FOUND => [
+        'name' => Exception::RULE_NOT_FOUND,
+        'description' => 'Rule with the requested ID could not be found.',
+        'code' => 404,
+    ],
+    Exception::RULE_ALREADY_EXISTS => [
+        'name' => Exception::RULE_ALREADY_EXISTS,
+        'description' => 'Domain already used.',
+        'code' => 409,
+    ],
+    Exception::RULE_VERIFICATION_FAILED => [
+        'name' => Exception::RULE_VERIFICATION_FAILED,
```

Comment:
Will rule always be coupled with a domain? When we add domain, I guess we'll bring back `DOMAIN_VERIFICATION_FAILED` should we still use it?

Reply:
I didnt get you. We removed concept of "Domains", we now have "Rules". "Domains" will be what you pay for monthly, and where you set DNS records.

What's your suggestion here? I cant see a problem with `RULE_VERIFICATION_FAILED `.